proposals/interface-types/working-notes/scenarios/unicode_count.md

```diff
@@ -77,36 +77,35 @@ that highlights that this is not a regular webAssembly function:
   
 (@interface implement (import "" "countCodes_")
   (param $ptr i32)(param $len i32) (result i32))
-  defer-scope
-    local.get $ptr
-    local.get $len
-    memory-to-string "memi"
-    call-import "countCodes"
-    u32-to-i32
-  end
+  local.get $ptr
+  local.get $len
+  string.from.utf8 "memi"
+  call-import "countCodes"
+  i32.from.u32
 )
 ```
 
-The heart of this adapter is the special adapter instruction `memory-to-string`
+The heart of this adapter is the special adapter instruction `string.from.utf8`
```

Comment:
In Instructions.md, we have `string.lift_memory` and `string.lower_memory`.  Is `string.from.utf8` and `utf8.from.string` proposed anywhere?

Reply:
Well, I am not much in favor of mentioning lower & lift in the instruction names.
I believe instruction names should state what they do, not their purpose. (This is especially true for low-level instruction names that could easily be repurposed.)
Also, I added utf8 as a signal that we may be supporting other encodings. This is where the encoding makes a difference.